crates/matrix-sdk-ui/src/notification_client.rs

```diff
@@ -322,25 +337,57 @@ impl NotificationClient {
                 }
             });
 
-        let cloned_notif = notification.clone();
+        // We'll only use this event if the room is in the invited state.
+        let raw_invite = Arc::new(Mutex::new(None));
+
         let target_event_id = event_id.to_owned();
+        let user_id = self.client.user_id().unwrap().to_owned();
+        let handler_raw_invite = raw_invite.clone();
+        let handler_raw_notification = raw_notification.clone();
         let stripped_member_handler =
             self.client.add_event_handler(move |raw: Raw<StrippedRoomMemberEvent>| async move {
+                let deserialized = match raw.deserialize() {
+                    Ok(d) => d,
+                    Err(err) => {
+                        warn!("failed to deserialize raw stripped room member event: {err}");
+                        return;
+                    }
+                };
+
+                trace!("received a stripped room member event");
+
+                // Try to match the event by event_id, as it's the most precise. In theory, we
+                // shouldn't receive it, so that's a first attempt.
                 match raw.get_field::<OwnedEventId>("event_id") {
                     Ok(Some(event_id)) => {
                         if event_id == target_event_id {
                             // found it! There shouldn't be a previous event before, but if there
                             // is, that should be ok to just replace it.
-                            *cloned_notif.lock().unwrap() = Some(RawNotificationEvent::Invite(raw));
+                            *handler_raw_notification.lock().unwrap() =
+                                Some(RawNotificationEvent::Invite(raw));
+                            return;
                         }
                     }
                     Ok(None) => {
-                        warn!("a room member event had no id");
+                        debug!("a room member event had no id");
                     }
                     Err(err) => {
-                        warn!("a room member event id couldn't be decoded: {err}");
+                        debug!("a room member event id couldn't be decoded: {err}");
                     }
                 }
+
+                // Try to match the event by membership and state_key for the current user.
+                if deserialized.content.membership == MembershipState::Invite
+                    && deserialized.state_key == user_id
```

Comment:
I'm not sure it's gonna work with the latest version of Ruma. Either you transform `user_id` into the type of `state_key`, or you extract the user ID from `state_key` with `state_key.user_id()`.

Reply:
This is rebased on main, so should be fine.

Reply:
(It's not a call room member event, but a general room member event.)

Reply:
Alright.